### PR TITLE
api v1 - config templates

### PR DESCRIPTION
**Pull request by Pajk**
I am not able to make all tests pass. Problem is with "should build pxe menu", could somebody help me with this?

It fails in render_safe with this error:

```
undefined method `new' for #<Class:0x00000002895d08> (NoMethodError)
/home/ppokorny/.gem/ruby/1.9.1/gems/safemode-1.0.1/lib/safemode.rb:43:in `initialize'
/home/ppokorny/Projects/ruby/foreman/lib/foreman/renderer.rb:8:in `new'
/home/ppokorny/Projects/ruby/foreman/lib/foreman/renderer.rb:8:in `render_safe'
/home/ppokorny/Projects/ruby/foreman/app/controllers/api/v1/config_templates_controller.rb:71:in `build_pxe_default'
...
/home/ppokorny/Projects/ruby/foreman/test/functional/api/v1/config_templates_controller_test.rb:86:in `block (2 levels) in <class:ConfigTemplatesControllerTest>'
/home/ppokorny/Projects/ruby/foreman/test/test_helper.rb:30:in `as_user'
/home/ppokorny/Projects/ruby/foreman/test/functional/api/v1/config_templates_controller_test.rb:85:in `block in <class:ConfigTemplatesControllerTest>'
....
```

### CODE REVIEW

**Comment by ohadlevy** [comment]
can you try with ruby 1.87 or 1.93 ?

**Comment by Pajk** [comment]
I will try it with 1.8. This error is with ruby 1.9.3p194.

File: app/controllers/api/v1/config_templates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      def destroy
+        process_response @config_template.destroy
+      end
+
+      api :GET, "/config_templates/build_pxe_default", "Change the default PXE menu on all configured TFTP servers"
+      def build_pxe_default
+        if (proxies = Subnet.all.map(&:tftp).uniq.compact).empty?
+          error_msg = "No TFTP proxies defined, can't continue"
+        end
+
```
Comment (ohadlevy) [inline app/controllers/api/v1/config_templates_controller.rb]: I wonder if this method doesn't need to move into the model, so we don't have to duplicate it.

**Comment by Pajk** [comment]
With 1.8.7 it looks ok. With 1.9.[2|3] it fails with mentioned error.

**Comment by Pajk** [comment]
@ohadlevy I moved build_pxe_default into the model but still using controller for rendering because I don't think it should be done in the model. It is also impossible without small hack to use url_for in models (it is needed in default_template_url method).

File: app/controllers/api/v1/config_templates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+module Api
+  module V1
+    class ConfigTemplatesController < V1::BaseController
+      include Foreman::Controller::AutoCompleteSearch
```
Comment (ohadlevy) [inline app/controllers/api/v1/config_templates_controller.rb]: not really required

**Comment by ohadlevy** [comment]
+1 I'll merge tomorrow

File: app/views/api/v1/config_templates/show.json.rabl [hunk 3]
```diff
@@ -0,0 +1,3 @@
+object @config_template
+
+attributes :name, :template, :snippet, :audit_comment, :template_kind_id, :template_combinations, :id
```
Comment (ohadlevy) [inline app/views/api/v1/config_templates/show.json.rabl]: should we shoe the combinations kind etc only if its not a snippet?

Comment (Pajk) [inline app/views/api/v1/config_templates/show.json.rabl]: You tell me. Which attributes are snippet-only and which  template-only?

**Comment by ohadlevy** [comment]
merged with minor modifications - thanks!

**Pull request closed without merging**